controllers/topology/util_test.go

```diff
@@ -21,9 +21,8 @@ import (
 
 	"github.com/pkg/errors"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
-	"sigs.k8s.io/cluster-api/internal/testtypes"
+	. "sigs.k8s.io/cluster-api/internal/testtypes"
```

Comment:
Moved from the old PR:

> @sbueringer sbueringer 4 days ago Member
> I'm not a fan of dot imports to be honest :/ (because of the "magic-factor")
> I understand why we want it for matchers, I'm not sure if we should also use it for our builder funcs.

 
> @killianmuldoon killianmuldoon 3 days ago Author Member
> Yeah - it's a compromise here from having one package with the two different functions. We > are using . imports across our testing for Gomega. Do you have strong feelings about this? I can definitely see both sides.

Reply:
@killianmuldoon It's fine for me to keep it, if that's the consensus. Overall I would like to be very careful with dot imports. Gomega is basically the only one I ever saw in Go and I think that's a good thing because it makes the code less readable.

An option could be to have the matchers and builders in separate packages and only import the matchers with a dot import (like the gomega matchers).

I would like to hear opinions from @vincepri and @fabriziopandini on that, though.

Reply:
I'm personally ok with this dot import given that it is our own "additional layer" on top of gomega/testing and it makes tests more readable. However, no problem to accept whatever we agree on.

Reply:
I definitely agree with having the dot imports for the matchers. It's more the builder funcs and everything else we currently have and will add in the future in the testtypes package.

Reply:
I definitely agree there's a danger here - dot-by-default isn't something we want for the entire testtypes package I don't think. We could have testtypes/matchers below it - other than worries about too many small packages is there any issue with doing it that way and only doing dot import on matchers?

Reply:
I don't have a concern about a separate package, only wondering if a slightly different structure would make more sense:
```bash
internal
  testtypes
  matchers
```

Or
```bash
internal
  test
    types
    matchers
```

If I understood correctly the matchers are an extension to gomega matchers for CAPI/Kubernetes so they are not really specific to `testtypes` they are just also part of our "test library"